pkg/chunked/compression_linux.go

```diff
@@ -254,14 +292,19 @@ func readZstdChunkedManifest(blobStream ImageSourceSeekable, tocDigest digest.Di
 		if err != nil {
 			return nil, nil, nil, 0, err
 		}
-		decodedTarSplit, err = decodeAndValidateBlob(tarSplit, tarSplitLengthUncompressed, toc.TarSplitDigest.String())
+		decodedTarSplit, err = openTmpFile(tmpDir)
 		if err != nil {
+			return nil, nil, nil, 0, err
+		}
+		if err := decodeAndValidateBlobToStream(tarSplit, decodedTarSplit, toc.TarSplitDigest.String()); err != nil {
+			decodedTarSplit.Close()
```

Comment:
I have not fully understood all the code but it seem many callers no call seek(0,0) on the tarsplit, would it makes sense to seek once after this here instead? Seems kinda unintuitive and easy to forget for a caller otherwise that they must seek the returned file?

Reply:
If the file is accessed multiple times, we need to call seek(0, 0) each time to read it fully from the beginning.  So we can't just call it once

Reply:
ack, maybe add a comment to this function that the returned file was already read and the caller must seek to the beginning if it tries to read it again.
Also I guess `tar-split blob or nil` should be changed to `tar-split file or nil`